src/components/input/bl-input.ts

```diff
@@ -239,7 +286,7 @@ export default class BlInput extends FormControlMixin(LitElement) {
           aria-describedby=${ifDefined(this.helpText ? "helpText" : undefined)}
           aria-errormessage=${ifDefined(this.checkValidity() ? undefined : "errorMessage")}
         />
-        <div class="icon">${icon}<bl-icon class="error-icon" name="alert"></bl-icon></div>
+        <div class="icon">${revealButton || icon}<bl-icon class="error-icon" name="alert"></bl-icon></div>
```

Comment:
With this implementation, it's not possible to see a custom icon next to password reveal button. So

```html
<bl-input type="password" icon="key"></bl-input>
```

will not work as expected. I suggest to have another wrapper for reveal button similar to `<div class="icon">` and don't touch the logic about custom icon here.